src/slicegpt/hf_utils.py

```diff
@@ -74,48 +74,55 @@ def wrapper(*args, **kwargs):
 
 @do_not_initialize
 def get_model_and_tokenizer(
-    model_path: str, uninitialized: bool = False, dtype: torch.dtype = torch.float16, token: str | bool | None = None
+    model_name: str,
+    model_path: str | None = None,
+    uninitialized: bool = False,
+    dtype: torch.dtype = torch.float16,
+    token: str | bool | None = None,
 ) -> tuple[ModelAdapter, PreTrainedTokenizerBase]:
     """Loads the model and the tokenizer from the given path."""
     if uninitialized:
         model_type = "uninitialized"
     else:
         model_type = "pretrained"
 
-    tokenizer = AutoTokenizer.from_pretrained(model_path, use_fast=True, token=token)
+    if not model_path:
+        # HF models can be downloaded using the name only, local models need to specify a path
+        model_path = model_name
 
-    logging.info(f"Loading {model_type} {model_path} model")
+    logging.info(f"Loading {model_type} {model_name} model from {model_path}")
 
-    if "facebook/opt" in model_path:
+    if model_name.startswith("facebook/opt"):
         if uninitialized:
-            config = OPTConfig.from_pretrained(model_path)
+            config = OPTConfig.from_pretrained(model_path, torch_dtype=dtype)
             model = UninitializedOPTForCausalLM(config)
             model = model.to(dtype=dtype)
         else:
             model = OPTForCausalLM.from_pretrained(model_path, torch_dtype=dtype)
             model.config.torch_dtype = dtype
+        tokenizer = AutoTokenizer.from_pretrained(model_path, use_fast=True, token=token)
         model_adapter = OPTModelAdapter(model)
-    elif "meta-llama" in model_path:
+    elif model_name.startswith("meta-llama/Llama-2"):
         if uninitialized:
-            config = LlamaConfig.from_pretrained(model_path, token=token)
+            config = LlamaConfig.from_pretrained(model_path, torch_dtype=dtype, token=token)
             model = UninitializedLlamaForCausalLM(config)
             model = model.to(dtype=dtype)
         else:
             model = LlamaForCausalLM.from_pretrained(model_path, torch_dtype=dtype, token=token)
             model.config.torch_dtype = dtype
-
+        tokenizer = AutoTokenizer.from_pretrained(model_path, use_fast=True, token=token)
```

Comment:
Nit: I think the tokenizer logic is common to all models so can now be pulled out of the `if/else` block, placed after it and have an `if` to add a pad token if needed.